src/bika/lims/content/bikasetup.py

```diff
@@ -1395,5 +1399,57 @@ def setInvalidationReasonRequired(self, value):
         if setup:
             setup.setInvalidationReasonRequired(value)
 
+    def getWorksheetLayout(self):
+        setup = api.get_senaite_setup()
+        # setup is `None` during initial site content structure installation
+        if setup:
+            return setup.getWorksheetLayout()
+        return self.getField("WorksheetLayout").default
+
+    def setWorksheetLayout(self, value):
+        setup = api.get_senaite_setup()
+        # setup is `None` during initial site content structure installation
+        if setup:
+            setup.setWorksheetLayout(value)
+
+    def getRestrictWorksheetUsersAccess(self):
+        setup = api.get_senaite_setup()
+        # setup is `None` during initial site content structure installation
+        if setup:
+            return setup.getRestrictWorksheetUsersAccess()
+        return self.getField("RestrictWorksheetUsersAccess").default
+
+    def setRestrictWorksheetUsersAccess(self, value):
+        setup = api.get_senaite_setup()
+        # setup is `None` during initial site content structure installation
+        if setup:
+            setup.setRestrictWorksheetUsersAccess(value)
+
+    def getRestrictWorksheetManagement(self):
```

Comment:
See comment above. Copy the value of this `RestrictWorksheetManagement` field from `bika_setup` to `setup` in the upgrade step.